src/com/google/enterprise/adaptor/testing/RecordingResponse.java

```diff
@@ -0,0 +1,222 @@
+// Copyright 2011 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.enterprise.adaptor.testing;
+
+import com.google.enterprise.adaptor.Acl;
+import com.google.enterprise.adaptor.Metadata;
+import com.google.enterprise.adaptor.Response;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A fake implementation of {@link Response} that simply records the
+ * values it receives.
+ */
+public class RecordingResponse implements Response {
+  // TODO(jlacey): Implement Response2.
```

Comment:
You've convinced me that we can require changes to src and that we will sufficiently advertise and assist with those changes. So I think we are both saying no Response2, now.